### ISSUE TITLE
[ENH] Standardize scalar scoring in pgmpy metrics

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#3579]
Metrics implemented in pgmpy (`pgmpy/metrics`) share the same callable interface but return types depend on the specific metric and can be one of: scalars, dictionaries, tuples, or DataFrames. This makes it difficult for generic callers (such as `CausalDiscovery.score`) and they need metric-specific handling to distinguish between different return types.

For example:

  ```python
  from pgmpy.datasets import load_dataset
  from pgmpy.metrics import SHD, AdjacencyConfusionMatrix, ImpliedCIs

  ds = load_dataset("linear_gaussian", n_samples=100, seed=42)
  g = ds.ground_truth

  SHD()(g, g)                                 # scalar
  AdjacencyConfusionMatrix()(g, g)             # dict
  ImpliedCIs(show_progress=False)(ds.data, g)  # DataFrame
  ```

  Proposed changes:

  - Add individual scalar metrics such as `AdjacencyPrecision`, `AdjacencyRecall`, and `RMSEA`, sharing existing computations.
  - Preserve existing report outputs.
  - Add an `output_type` tag (`"scalar"` or `"report"`) that reflects the configured instance. Retain `lower_is_better`, with `None` when automatic ranking is inappropriate.
  - Update the metric documentation and extension template, and test agreement between new scalar metrics and existing statistics.

Proposed API example:

  ```python
  from pgmpy.metrics import AdjacencyPrecision, CorrelationScore

  metric = AdjacencyPrecision()
  metric(g, g)                           # float
  metric.get_tag("output_type")          # "scalar"
  metric.get_tag("lower_is_better")      # False

  AdjacencyConfusionMatrix().get_tag("output_type")              # "report"
  CorrelationScore(return_summary=True).get_tag("output_type")   # "report"
  ```

  This builds on [sktime’s tagged metric objects](https://www.sktime.net/docs/api-reference/sktimeperformance-metricsforecastingmeanabsoluteerror/) and supports named scalar evaluation similar to [sklearn’s multi-metric scoring](https://scikit-learn.org/stable/modules/model_evaluation.html#using-multiple-metric-evaluation).

**Comment by Indrapal-70** [comment]
HI @ankurankan I would like to work on this issue , Could you please assign this to me ?

**Comment by Kushagra651** [comment]
Hi @ankurankan, I'd like to work on this!

I've gone through pgmpy/metrics/ and the proposed API makes sense to me. Before I start, a couple of quick questions:
Should the tag system follow the same _tags dict + get_tag() pattern as in BaseStructureScore, or is there a separate base for metrics I should extend?
For AdjacencyPrecision and AdjacencyRecall — should these reuse the existing AdjacencyConfusionMatrix computation internally?

I'll post a design sketch here before opening a PR, same as I did for #3489.

**Comment by adaniel105** [comment]
Hello @ankurankan, if this issue is still available I can pick it up. A proposed implementation resolving this can go as follows:

- For scalar metrics that inherit from `AdjacencyConfusionMatrix` i.e  `AdjacencyPrecision` and `AdjacencyRecall`, we can calculate the individual scalar metric by passing `cm = AdjacencyConfusionMatrix(metrics=["metric"])` as follows (if necessary, we can memoize the computation of TN, FN, TP and FP with `joblib` to avoid recomputation of these metrics on identical graphs everytime `AdjacenctConfusionMatrix` is called by individual scalar functions):

```python
class AdjacencyPrecision(BaseSupervisedMetric):
    _tags = {
        "name": "adjacency_precision",
        "requires_true_graph": True,
        "requires_data": False,
        "lower_is_better": False,
        "is_symmetric": False,
        "supported_graph_types": (DAG, PDAG),
        "output_type": "scalar"
    }
    def calculate(self, true_causal_graph, est_causal_graph):
        cm = AdjacencyConfusionMatrix(metrics=["precision"])
        precision = cm.evaluate(true_causal_graph, est_causal_graph)
        return precision
```

This preserves the existing report returned by `AdjacencyConfusionMatrix` and mitigates code redundancy, keeping its `_tags` dict as:

```python
    _tags = {
        "name": "adjacency_confusion_matrix",
        "requires_true_graph": True,
        "requires_data": False,
        "lower_is_better": False,
        "is_symmetric": False,
        "supported_graph_types": (DAG, PDAG),
        "output_type": "report"
    }
```



- For the RMSEA calculation, we can leave the specific computation of chi-square and degree of freedom to the individual methods calling the function and implement a general `compute_rmsea` function like so:

```python
class RMSEA(BaseUnsupervisedMetric):
    _tags = {
        "name": "RMSEA",
        "requires_true_graph": False,
        "requires_data": False,
        "lower_is_better": True,
        "is_symmetric": False,
        "supported_graph_types": (DAG,),
        "output_type": "scalar"
    }

    def _compute_rmsea(self ,chi_squared, deg_of_freedom, X):
        if X <= 1 or deg_of_freedom <= 0:
            return np.nan
        return np.sqrt(max((chi_squared - deg_of_freedom) / (deg_of_freedom * (X - 1)), 0))
```

We can then just add the `output_type` metrics to `_tags` as done in the above code snippets, which are already enumerated and retrieved by the skbase's BaseObject `get_tags` / `get_tag` methods , which both `BaseSupervisedMetric` and `BaseUnsupervisedMetric` inherit from, leading to a seamless implementation of the proposed API example.

**Comment by Shweta-nasc** [comment]
Hi @ankurankan, I’d like to work on this issue. Could you please assign it to me? I’d be happy to take this up. Thanks!

**Comment by Jayant-kernel** [comment]
Hi @ankurankan 
, I’d like to work on this issue if it’s still available.
 I’ll review the existing metric implementations, add the scalar metrics and `output_type`/`lower_is_better` tags while preserving current report outputs, then update the documentation and add tests for consistency. 
I’ll share the approach before opening a PR.
